### ISSUE TITLE
Searching for 'search' returns no results in Safari on Mac and iPhone

### ISSUE THREAD

**Issue by martinbira** [squidfunk/mkdocs-material#915]
## Description

When searching for the word `search` no results are found for Safari users on Mac and iPhone. Tested on iPhone X using Safari, Mac using Safari and Chrome, as well as Win 10 with Microsoft Edge, Chrome, Firefox, and Vivaldi browsers.

### Expected behavior

Pages where the word `search` should appear in the search results.

### Actual behavior

No results found for the word `search` in the search results.

### Steps to reproduce the bug

1. Have page with the word `search` in the `search_index.json`
2. Search for `search`.

### Package versions

* Python: 3.6.1
* MkDocs: 1.0.4
* Material: 3.0.6

### System information

* OS: OS X
* Browser: Safari

**Comment by squidfunk** [comment]
Hmm, works in the official docs:

![bildschirmfoto 2018-11-09 um 14 53 16](https://user-images.githubusercontent.com/932156/48266313-4d85a480-e42f-11e8-89bc-00b7b28a56c7.png)

Safari, macOS

**Comment by martinbira** [comment]
So I've been going over the issue with two of our Mac users here and they both report the same console errors in Safari, which may have something to do with the indexing. We've used both our full search index straight from `mkdocs build` and our slightly modified index with the same results.

```
application.5e60981f.js 
Line 2434:77 undefined is not an object (evaluating 'm_index')
```
![safari_console_error](https://user-images.githubusercontent.com/22232681/48408790-7a410100-e73a-11e8-828e-0c11ef260194.png)

**Comment by squidfunk** [comment]
This error occurs in the source of `lunr.js`, not Material. It could, however, be related to some invalid value that is passed to `lunr.js`. Could you try narrowing it further down?

**Comment by martinbira** [comment]
This is not reproducable on mkdocs.org. We see it on Material directly when entering a key, such as `j` as in the following screenshot.

![safari_console_official_page](https://user-images.githubusercontent.com/22232681/48467929-b338ae80-e7ea-11e8-8d9d-f3d9ffde3cc2.png)

**Comment by squidfunk** [comment]
Awesome, I can also reproduce it now. Thanks for investigating. I will debug it when I find the time, most likely by the beginning of next week. If somebody else wants to go ahead, feel free to PR!

**Comment by martinbira** [comment]
It sure looks similiar after a quick glance.

**Comment by squidfunk** [comment]
Indeed, it looks like it's related to the issue posted by @max-ci. I skimmed through the issue and it seems that 2.3.1 includes a fix for this, so we could bump `lunr.js` to the current version and try if that fixes the problem.

**Comment by squidfunk** [comment]
I upgraded to Lunr `2.3.5` and it seems fixed. I will issue a release within a few minutes.

**Comment by squidfunk** [comment]
Released as part of 3.1.0
